core/src/main/java/com/flipkart/varadhi/core/cluster/ConsumerApiFactory.java

```diff
@@ -0,0 +1,5 @@
+package com.flipkart.varadhi.core.cluster;
+
+public interface ConsumerApiFactory {
+    ConsumerApi getConsumerProxy(String consumerId);
```

Comment:
```suggestion
    ConsumerApi getInstance(String consumerId);
```
I think we dont have to resort to "proxy" notion. 
Usually, factory classes have getInstance/createInstance methods and they work fine here as well.